configure.ac

```
@@ -442,6 +442,10 @@ if test "x$enable_werror" = "xyes"; then
 
   if test x$suppress_external_warnings != xno ; then
     AX_CHECK_COMPILE_FLAG([-Werror=documentation],[ERROR_CXXFLAGS="$ERROR_CXXFLAGS -Werror=documentation"],,[[$CXXFLAG_WERROR]])
+    AX_CHECK_COMPILE_FLAG([-Werror=zero-as-null-pointer-constant],[
+      WARN_CXXFLAGS="$WARN_CXXFLAGS -Werror=zero-as-null-pointer-constant"
+      AC_DEFINE(HAVE_ZERO_AS_NULL_POINTER_CONSTANT, 1, [Define this symbol if -Wzero-as-null-pointer-constant is available])
+    ],,[[$CXXFLAG_WERROR]])
```

One pair of `[]` is redundant:
```suggestion
    AX_CHECK_COMPILE_FLAG([-Werror=zero-as-null-pointer-constant], [
      WARN_CXXFLAGS="$WARN_CXXFLAGS -Werror=zero-as-null-pointer-constant"
      AC_DEFINE(HAVE_ZERO_AS_NULL_POINTER_CONSTANT, 1, [Define this symbol if -Wzero-as-null-pointer-constant is available])
    ], [], [$CXXFLAG_WERROR])
```

---

I'm an autoconf novice - I take it we're not concerned about macro conflicts with the CXXFLAG_WERROR as its contents are of a different form than uppercase macro definitions?

> When you use the same text in a macro argument, you must therefore have an extra quotation level (since one is stripped away by the macro substitution). In general, then, it is a good idea to use double quoting for all literal string arguments, either around just the problematic portions, or over the entire argument:

>     AC_MSG_WARN([[AC_DC] stinks  --Iron Maiden])
>     AC_MSG_WARN([[AC_DC stinks  --Iron Maiden]])
https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/Autoconf-Language.html#Autoconf-Language